app/models/concerns/obstructed.rb

```diff
@@ -0,0 +1,70 @@
+# In queen.rb, rook.rb, bishop.rb, pawn.rb, I added "include Obstructed"

```

Comment:
Design issue: I'd argue that these methods should be in the Pieces model rather than abstracted out as a concern. I think of concerns as what you'd use when you want to include functionality that can't or shouldn't be inherited. 

Then, instead of having to pass in start_x and start_y, they would already be available as member variables of Piece.